pkg/apis/contrail/v1alpha1/cassandra_types.go

```diff
@@ -240,7 +240,21 @@ func (c *Cassandra) PodIPListAndIPMapFromInstance(instanceType string, request r
 
 // SetInstanceActive sets the Cassandra instance to active.
 func (c *Cassandra) SetInstanceActive(client client.Client, activeStatus *bool, sts *appsv1.StatefulSet, request reconcile.Request) error {
-	return SetInstanceActive(client, activeStatus, sts, request, c)
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: sts.Name, Namespace: request.Namespace},
+		sts); err != nil {
+		return err
+	}
+	active := false
+	var acceptableReadyReplicaCnt = *sts.Spec.Replicas/2 + 1
+	if sts.Status.ReadyReplicas >= acceptableReadyReplicaCnt {
+		active = true
+	}
+
+	*activeStatus = active
+	if err := client.Status().Update(context.TODO(), c); err != nil {
```

Comment:
I think that you can just return directly value from Update method.
'return client.Status().Update()

Reply:
fixed